File: app/src/main/java/com/justnik/mirea_viewbinding_lesson/MainActivity.java
package com.justnik.mirea_viewbinding_lesson;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.justnik.mirea_viewbinding_lesson.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        setClickListener();
    }

    private void setClickListener() {
        binding.bOpenActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, ExampleActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/justnik/mirea_viewbinding_lesson/MainActivity.kt
package com.justnik.mirea_viewbinding_lesson

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import by.kirich1409.viewbindingdelegate.viewBinding
import com.justnik.mirea_viewbinding_lesson.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity(R.layout.activity_main) {

    private val binding: ActivityMainBinding by viewBinding()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setClickListener()
    }

    private fun setClickListener(){
        binding.bOpenActivity.setOnClickListener {
            val intent = Intent(this@MainActivity, ExampleActivity::class.java)
            startActivity(intent)
        }
    }
}